finalitygadget/finalitygadget.go

```diff
@@ -92,8 +92,8 @@ func NewFinalityGadget(cfg *config.Config, db db.IDatabaseHandler, logger *zap.L
 
 	lastProcessedHeight := uint64(0)
 	latestBlock, err := db.QueryLatestFinalizedBlock()
-	if err != nil {
-		return nil, err
+	if err != nil && !errors.Is(err, types.ErrBlockNotFound) {
```

Comment:
why do we need this special check of `ErrBlockNotFound`?

Reply:
It is useful when the latest block hasn't been stored yet
```
	if err != nil {
		// If no latest block has been stored yet, return nil
		if errors.Is(err, types.ErrBlockNotFound) {
			return nil, nil
		}
		bb.logger.Error("Error getting latest block", zap.Error(err))
		return nil, err
	}
```